### DISCUSSION TITLE
How to use hide side functionalities with mdformat github action

### DISCUSSION THREAD

**Discussion by LuigiCerone** [squidfunk/mkdocs-material#5331] [Q&A] [answered]
Hello!

I'm using [Hiding sidebar functionalities](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#hiding-the-sidebars) in a Python repo that has also pre-commit configured with [mdformat](https://github.com/executablebooks/mdformat).
The problem is that this:

```md
---
hide:
  - navigation
---
```

is reformatted into this:

```md
______________________________________________________________________

hide:

- navigation

______________________________________________________________________
```

and after that it is not recognized by mkdocs and therefore it doesn't work. Is there a way to make this theme support the mdformat syntax? Thanks :)

**Comment by kamilkrzyskow** [comment, marked as answer]
The header meta data of a Markdown file is called front matter.

The issue is that your mdformat tool doesn't support it.

https://github.com/executablebooks/mdformat/issues/332

In the linked issue thread some mdformat-mkdocs plugin collection is mentioned you could install and add such support but I did not try it myself :v:
